pkg/controller/propagator/propagation.go

```diff
@@ -350,3 +389,91 @@ func (r *ReconcilePolicy) handleDecision(instance *policiesv1.Policy, decision a
 	}
 	return nil
 }
+
+// a helper to quickly check if there are any templates in any of the policy templates
+func policyHasTemplates(instance *policiesv1.Policy) bool {
+	for _, policyT := range instance.Spec.PolicyTemplates {
+		if templates.HasTemplate(policyT.ObjectDefinition.Raw, templateCfg.StartDelim) {
+			return true
+		}
+	}
+	return false
+}
+
+// iterates through policy definitions  and  processes hub templates
+// a special  annotation policy.open-cluster-management.io/trigger-update is used to trigger reprocessing of the
+// templates and ensuring that the replicated-policies in cluster is updated only if there is a change.
+// this annotation is deleted from the replicated policies and not propagated to the cluster namespaces.
+
+func (r *ReconcilePolicy) processTemplates(replicatedPlc *policiesv1.Policy, decision appsv1.PlacementDecision, rootPlc *policiesv1.Policy) error {
+
+	reqLogger := log.WithValues("Policy-Namespace", rootPlc.GetNamespace(), "Policy-Name", rootPlc.GetName(), "Managed-Cluster", decision.ClusterName)
+	reqLogger.Info("Processing Templates..")
+
+	templateCfg.LookupNamespace = rootPlc.GetNamespace()
+	tmplResolver, err := templates.NewResolver(kubeClient, kubeConfig, templateCfg)
+	if err != nil {
+		reqLogger.Error(err, "Error instantiating template resolver")
+		panic(err)
+	}
+
+	//A policy can have multiple policy templates within it, iterate and process each
+	for _, policyT := range replicatedPlc.Spec.PolicyTemplates {
+
+		if templates.HasTemplate(policyT.ObjectDefinition.Raw, templateCfg.StartDelim) {
```

Comment:
Optional: It'd be easier to read if you did this instead so you can dedent the rest of the code and the reader does not need to keep in mind of the context of the previous if statement.

```go
if !templates.HasTemplate(policyT.ObjectDefinition.Raw, templateCfg.StartDelim) {
    continue
}
```